File: Counting Sort/Counting Sort.py
def counting_sort(arr):
    max_val = max(arr)
    count = [0] * (max_val + 1)
    output = [0] * len(arr)
    for i in arr:
        count[i] += 1
    for i in range(1, max_val + 1):
        count[i] += count[i - 1]
    for i in reversed(arr):
        output[count[i] - 1] = i
        count[i] -= 1
    return output

File: Counting Sort/Counting Sort.java
void countingSort(int[] arr) {
    int max = arr[0];
    for (int i = 1; i < arr.length; i++) if (arr[i] > max) max = arr[i];
    int[] count = new int[max + 1];
    int[] output = new int[arr.length];
    for (int i : arr) count[i]++;
    for (int i = 1; i <= max; i++) count[i] += count[i - 1];
    for (int i = arr.length - 1; i >= 0; i--) {
        output[count[arr[i]] - 1] = arr[i];
        count[arr[i]]--;
    }
    for (int i = 0; i < arr.length; i++) arr[i] = output[i];
}
